src/base/registrations/registrar.ts

```diff
@@ -109,34 +136,32 @@ export async function getAvatar(name: string, config: Config, resolver?: EnsReso
   return resolver.getText('avatar');
 }
 
-export async function getSeedHost(name: string, config: Config, resolver?: EnsResolver | null): Promise<string | null> {
+export async function getAnchorsAccount(name: string, config: Config, resolver?: EnsResolver | null): Promise<string | null> {
   name = name.toLowerCase();
 
   resolver = resolver ?? await config.provider.getResolver(name);
   if (! resolver) {
     return null;
   }
-  return resolver.getText('eth.radicle.seed.host');
+  return resolver.getText('eth.radicle.anchors');
 }
 
-export async function getSeedId(name: string, config: Config, resolver?: EnsResolver | null): Promise<string | null> {
+export async function getSeed(name: string, config: Config, resolver?: EnsResolver | null): Promise<Seed | null> {
   name = name.toLowerCase();
 
   resolver = resolver ?? await config.provider.getResolver(name);
   if (! resolver) {
     return null;
   }
-  return resolver.getText('eth.radicle.seed.id');
-}
 
-export async function getAnchorsAccount(name: string, config: Config, resolver?: EnsResolver | null): Promise<string | null> {
-  name = name.toLowerCase();
+  const [host, id, git, api] = await Promise.all([
+    resolver.getText('eth.radicle.seed.id'),
+    resolver.getText('eth.radicle.seed.host'),
+    resolver.getText('eth.radicle.seed.git'),
+    resolver.getText('eth.radicle.seed.api'),
+  ]);
 
-  resolver = resolver ?? await config.provider.getResolver(name);
-  if (! resolver) {
-    return null;
-  }
-  return resolver.getText('eth.radicle.anchors');
+  return new Seed(id, host, git, api);
```

Comment:
This will be an issue, since `getText` could eventually return a `null` value, but in the creation of the `Seed` class we accept only `string | undefined`

Maybe we could do a map similar to.
```ts
const [host, id, git, api] =
    seedPromise.map(r => r.status == "fulfilled" && r.value ? r.value : undefined);
```

Reply:
Hmm but what is the issue exactly? It seems like typescript is fine compiling this, and the code will work since we check for truthy-ness only?

Reply:
I guess you mean when we upgrade to ethers 6? I think we can just deal with that issue when/if the time comes..

Reply:
Yeah I don't know why typescript compiles fine for you, for me it throws a typecheck error, that it can't assign a `string | null` value to `string | undefined`..
But if you want to go ahead, since the CI also doesn't throw errors and the build seems to work, why not :smiley: